### PR TITLE
[API Documentation] Add basic documentation for chrome.scripting

### LINKED ISSUE #194: Enhance the scripting API reference docs
Once the generated docs for the scripting API are included, need to update this by:

- Add enhanced introductory text to provide context and more how-to focus
- Add xrefs to the new API reference from places that mention use of the scripting API.

### PR DESCRIPTION
**Pull request by rdcronin**
Add basic documentation for the MV3 chrome.scripting API, including a
handful of (very) small examples.

Fixes #194

### CODE REVIEW

File: site/en/docs/extensions/reference/scripting/index.md [hunk 1]
```diff
@@ -2,9 +2,191 @@
...
+`scripting.insertCSS()`; you cannot execute a string via
+`scripting.executeScript()`.
+
+```js
+const css = 'body { background-color = "red"; }';
+const tabId = getTabId();
+chrome.scripting.insertCSS(
+    {
+      target: {tabId: tabId},
+      css: css,
```
Comment (samthor) [inline site/en/docs/extensions/reference/scripting/index.md:158]: nit: you can write this as just `css,`

File: site/en/docs/extensions/reference/scripting/index.md [hunk 2]
```diff
@@ -2,9 +2,191 @@
...
+  return document.title;
+}
+const tabId = getTabId();
+chrome.scripting.executeScript(
+    {
+      target: {tabId: tabId, allFrames: true},
+      function: getTitle,
+    },
+    (injectionResults) => {
+      for (const frameResult : injectionResults)
```
Comment (samthor) [inline site/en/docs/extensions/reference/scripting/index.md:181]: I think you've been writing too much C++: `:` should probably be `of`.

**Review by samthor — APPROVED** [review]
I'm not a tech writer but I am generally happy with this.

cc @awfuchs

File: site/en/docs/extensions/reference/scripting/index.md (hunk 2 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/scripting/index.md:181]: Guilty!  Fixed.

File: site/en/docs/extensions/reference/scripting/index.md (hunk 1 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/scripting/index.md:158]: yep, same for tabId, too.  I personally find this a bit more clear (though longer).  I'm not sure if that's a common sentiment, or me being old ; )  Simeon or Andy, any preference?

File: site/en/docs/extensions/reference/scripting/index.md [hunk 3]
```diff
@@ -2,9 +2,191 @@
...
+  ...
+}
+```
+
+## Usage
+
+You can use the `chrome.scripting` API to inject JavaScript and CSS into
+websites, similar to what can be achieved with
+[content scripts](contentScripts). With the `chrome.scripting` API, extensions
+can make decisions at runtime.
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/scripting/index.md:25]: Suggestion:
You can use the `chrome.scripting` API to inject JavaScript and CSS into
websites. This is similar to what you can do with
[content scripts](contentScripts), but by using the `chrome.scripting` API, extensions can make decisions at runtime.

File: site/en/docs/extensions/reference/scripting/index.md [hunk 4]
```diff
@@ -2,9 +2,191 @@
...
+
+### Injection Targets
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/scripting/index.md:27]: Here and below, use sentence case for titles and headings. Thus:

### Injection targets

https://developers.google.com/style/capitalization#capitalization-in-titles-and-headings

File: site/en/docs/extensions/reference/scripting/index.md [hunk 5]
```diff
@@ -2,9 +2,191 @@
...
+const tabId = getTabId();
+chrome.scripting.executeScript(
+    {
+      target: {tabId: tabId, allFrames: true},
+      files: ['script.js'],
+    },
+    () => { ... });
+```
+
+You can also inject in specific frames of a tab by specifying individual frame
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/scripting/index.md:58]: inject into

File: site/en/docs/extensions/reference/scripting/index.md [hunk 6]
```diff
@@ -2,9 +2,191 @@
...
+
+You can use the `chrome.scripting` API to inject JavaScript and CSS into
+websites, similar to what can be achieved with
+[content scripts](contentScripts). With the `chrome.scripting` API, extensions
+can make decisions at runtime.
+
+### Injection Targets
+
+You can use the `InjectionTarget` to specify an appropriate target to inject
+JavaScript or CSS into.
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/scripting/index.md:30]: Suggestion:
Use the `target` parameter to specify a target to inject
JavaScript or CSS into.

(Note I prefer to call a parameter by its name and not its type)

File: site/en/docs/extensions/reference/scripting/index.md [hunk 7]
```diff
@@ -2,9 +2,191 @@
...
+const frameIds = [frameId1, frameId2];
+chrome.scripting.executeScript(
+    {
+      target: {tabId: tabId, frameIds: frameIds},
+      files: ['script.js'],
+    },
+    () => { ... });
+```
+
+Note that you cannot specify both the `frameIds` and `allFrames` properties.
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/scripting/index.md:73]: Suggestion:
!!!.aside
You cannot specify both the `frameIds` and `allFrames` properties.
!!!

File: site/en/docs/extensions/reference/scripting/index.md [hunk 8]
```diff
@@ -2,9 +2,191 @@
...
+      files: ['script.js'],
+    },
+    () => { ... });
+```
+
 !!!.aside.warning
-This documentation is in active development.
-We're planning to expand the introduction section to provide more context and guidance on how to use this API.
-This work is tracked in issue [#194](https://github.com/GoogleChrome/developer.chrome.com/issues/194).
+Note: Currently, a maximum of a single file is supported.
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/scripting/index.md:97]: Either:
!!!.aside.aside--warning //This is the syntax you want, but warning is too strong
Warning: Currently...

Or:
!!!.aside
Note: Currently...

File: site/en/docs/extensions/reference/scripting/index.md [hunk 9]
```diff
@@ -2,9 +2,191 @@
...
+      target: {tabId: tabId},
+      function: changeBackgroundColor,
+    },
+    () => { ... });
+```
+
+You can work around this by using the [Storage API](storage) or by
+[passing messages](messaging).
+
+The Chromium team is working on providing more capabilities in the future.
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/scripting/index.md:143]: I'm not a fan of including predictions like this in reference docs...

File: site/en/docs/extensions/reference/scripting/index.md [hunk 10]
```diff
@@ -2,9 +2,191 @@
...
+
+#### Runtime Strings
+
+If injecting CSS within a page, you can also specify a string to be used in the
+`css` property. Note that this option is only available for
+`scripting.insertCSS()`; you cannot execute a string via
+`scripting.executeScript()`.
```
Comment (awfuchs) [inline site/en/docs/extensions/reference/scripting/index.md:150]: Suggestion:
If injecting CSS within a page, you can also specify a string to be used in the
`css` property. This option is only available for
`scripting.insertCSS()`; you can't execute a string using
`scripting.executeScript()`.

File: site/en/docs/extensions/reference/scripting/index.md (hunk 4 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/scripting/index.md:27]: Done

File: site/en/docs/extensions/reference/scripting/index.md (hunk 1 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/scripting/index.md:158]: Correct - using object declaration shorthand, the property is assigned to that variable's value.  This has been around for awhile, but I don't see it as much, and don't find it quite as clear.  But I'm not sure if that's just me not writing enough JS code ; )

File: site/en/docs/extensions/reference/scripting/index.md (hunk 3 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/scripting/index.md:25]: Done

File: site/en/docs/extensions/reference/scripting/index.md (hunk 5 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/scripting/index.md:58]: Done

File: site/en/docs/extensions/reference/scripting/index.md (hunk 6 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/scripting/index.md:30]: Done

File: site/en/docs/extensions/reference/scripting/index.md (hunk 7 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/scripting/index.md:73]: Done

File: site/en/docs/extensions/reference/scripting/index.md (hunk 8 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/scripting/index.md:97]: Updated to !!!.aside

File: site/en/docs/extensions/reference/scripting/index.md (hunk 9 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/scripting/index.md:143]: I don't feel strongly.  Removed.

File: site/en/docs/extensions/reference/scripting/index.md (hunk 10 above)
Comment (rdcronin) [reply site/en/docs/extensions/reference/scripting/index.md:150]: Done

File: site/en/docs/extensions/reference/scripting/index.md (hunk 1 above)
Comment (awfuchs) [reply site/en/docs/extensions/reference/scripting/index.md:157]: I think the way it is at lines 156-157 here is consistent with the way we're doing this elsewhere in the docs.  Fine as is.

**Review by awfuchs — APPROVED** [review]

**Pull request merged**